application/src/test/java/uk/nhs/hee/tis/revalidation/service/RecommendationTisStatusUpdateServiceTest.java

```diff
@@ -0,0 +1,206 @@
+  @Mock
+  private DoctorsForDBRepository doctorsForDBRepository;
+  @Mock
+  private SnapshotRepository snapshotRepository;
+  @Mock
+  private RecommendationService recommendationService;
+  @Mock
+  private SnapshotService snapshotService;
+
+  private String recommendationId;
+  private RecommendationGmcOutcome recommendationGmcOutcome;
+  private String firstName;
+  private String lastName;
+  private LocalDate submissionDate;
+  private LocalDate actualSubmissionDate;
+  private LocalDate dateAdded;
+  private UnderNotice underNotice;
+  private String sanction;
+  private String designatedBodyCode;
+  private RecommendationStatus status;
+  private String admin1;
+  private Snapshot snapshot;
+  private Recommendation recommendation;
+  private String gmcId;
+
+  @BeforeEach
+  public void setup() {
+    recommendationId = faker.lorem().characters(10);
+    recommendationGmcOutcome = RecommendationGmcOutcome.APPROVED;
+    firstName = faker.name().firstName();
+    lastName = faker.name().lastName();
+    status = COMPLETED;
+    submissionDate = LocalDate.now();
+    actualSubmissionDate = LocalDate.now();
+    dateAdded = LocalDate.now();
+    underNotice = faker.options().option(UnderNotice.class);
+    sanction = faker.lorem().characters(2);
+    designatedBodyCode = faker.lorem().characters(7);
+    admin1 = faker.internet().emailAddress();
+    gmcId = faker.number().digits(7);
+    recommendation = buildRecommendation(gmcId, recommendationId, status, REVALIDATE,
+        recommendationGmcOutcome);
+    snapshot = buildSnapshot(gmcId);
+  }
+
+  @Test
+  void shouldUpdateRecommendationAndTisStatus() throws Exception {
+
+    final var recommendationStatusCheckDto = RecommendationStatusCheckDto.builder()
+        .gmcReferenceNumber(gmcId)
+        .recommendationId(recommendationId)
+        .outcome(recommendationGmcOutcome)
+        .build();
+
+    final var doctorsForDB = buildDoctorForDB(gmcId);
+    when(doctorsForDBRepository.findById(gmcId)).thenReturn(of(doctorsForDB));
+
+    when(recommendationRepository.findById(recommendationId))
+        .thenReturn(Optional.of(buildRecommendation(gmcId, recommendationId, status, REVALIDATE,
+            UNDER_REVIEW)));
+
+    when(recommendationService.getRecommendationStatusForTrainee(gmcId)).thenReturn(status);
+
+    snapshotRepository.save(snapshot);
+
+    recommendationTisStatusUpdateService
+        .updateRecommendationAndTisStatus(recommendationStatusCheckDto);
+
+    verify(doctorsForDBRepository).save(doctorCaptor.capture());
+    assertThat(doctorCaptor.getValue().getDoctorStatus(), is(COMPLETED));
+
+    verify(recommendationRepository).save(recommendationCaptor.capture());
+    assertThat(recommendationCaptor.getValue().getRecommendationStatus(),
+        is(COMPLETED));
+    assertThat(recommendationCaptor.getValue().getOutcome(), is(RecommendationGmcOutcome.APPROVED));
+
+    verify(snapshotRepository).save(any());
```

Comment:
It's worth using a Captor here too and as Doris asked, writing another test case verifying that this is not called when a snapshot already exists.

Reply:
now calling the `snapshotService` and using a captor on the argument